pkg/interface/link/src/js/components/root.js

```diff
@@ -94,7 +99,7 @@ export class Root extends Component {
               )
             }}
           />
-          <Route exact path="/~link/(popout)?/:ship/:channel/:page/:index/(comments)?/:commentpage?"
+          <Route exact path="/~link/(popout)?/:ship/:channel/:page/:index/:encodedUrl/(comments)?/:commentpage?"
```

Comment:
So, this looks like this:

`~link/~/default/0/0/aHR0cDovLzRjaGFuLm9yZw==`

The `/0/0` there presents the same issue as the previous route — that is, that the permalink is a bit ephemeral. Maybe just strip the `:index` here.

Reply:
The permalink doesn't care about the `:index` here. But it's still used for the "click nth link from the list" flow, because I didn't completely overhaul the state structure. (iow, the "access by url" logic doesn't really know where to put the submission in the current state (because pages), so it just doesn't put it anywhere outside itself. (And state modification is considered a no-no outside of reducers anyway?))

Reply:
Going to `~link/~/default/0/2/aHR0cDovLzRjaGFuLm9yZw==` crashes, unable to find the `time` of `undefined`, so it seems to still pull its information from the `nth-link from list` flow, not the `:encodedUrll`?

Reply:
Cannot reproduce, works fine for me. Are you running an older version of `/app/link-view` or `/lib/link` maybe?

![no-repro](https://user-images.githubusercontent.com/3829764/73965316-3a080880-4914-11ea-9515-6b23b7f5d816.gif)

Reply:
Nope — I mean to change the second number in the URL and go directly to it. If that index isn't on the page (say, `19` in your screencap's case) it'll crash.

Reply:
In this case imagine someone loads their browser history and clicks a page that supposedly corresponds to the permalink in question (after all, that's why we even have the encodedUrl param, isn't it?), it'll load some random link that corresponds to that index on that page.

Reply:
Oh I see, can repro now. I'll work to fix this!

(But also, some quick playing around shows it _does_ work if I navigate to a url with any arbitrary _in-range_ index number. And since the bounds there only ever _increase_, your case _shouldn't_ happen in practice?)

Reply:
We should be defensive and never allow a crash to happen if we can help it, right?

Reply:
As said,

> I'll work to fix this!

Just wanted to clarify the limits of the issue & my implementation.

> seems like we need to redo that entire flow if we're going to actually solve this route problem since it always pulls from index; seemingly disregarding the permanent, encoded URL param

We don't, because as described above, it can work, _does_ work, and only needs a little more cleaning up elsewhere I'm pretty sure. Will report back after working on this.